internal/component/database_observability/lexer_test.go

```diff
@@ -309,3 +311,177 @@ func TestPgSqlParser_ExtractTableNames(t *testing.T) {
+		},
+		{
+			name:     "UPDATE statement",
+			query:    "UPDATE users SET name = 'John' WHERE id = 1",
+			expected: true,
+		},
+		{
+			name:     "SELECT with update in string",
+			query:    "SELECT * FROM users WHERE status = 'update_pending'",
+			expected: false,
+		},
+		{
+			name:     "DELETE statement",
+			query:    "DELETE FROM users WHERE id = 1",
+			expected: true,
+		},
+		{
+			name:     "SELECT with delete in table name",
+			query:    "SELECT * FROM delete_log",
+			expected: false,
+		},
+		{
+			name:     "plain SELECT statement",
+			query:    "SELECT * FROM users WHERE name = 'John'",
+			expected: false,
+		},
+		{
+			name:     "complex SELECT with joins",
+			query:    "SELECT u.name, p.title FROM users u JOIN posts p ON u.id = p.user_id",
+			expected: false,
+		},
+		{
+			name:     "SELECT with reserved word in WHERE clause string",
+			query:    "SELECT * FROM users WHERE description LIKE '%CREATE%'",
+			expected: false,
+		},
+		{
+			name:     "DROP statement",
+			query:    "DROP TABLE users",
+			expected: true,
+		},
+		{
+			name:     "SELECT with drop in column alias",
+			query:    "SELECT name AS drop_reason FROM users",
+			expected: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := ContainsReservedKeywords(tt.query, reservedWords, sqllexer.DBMSMySQL)
+			if result != tt.expected {
+				t.Errorf("Expected %v, got %v for query: %s", tt.expected, result, tt.query)
+			}
+		})
+	}
+}
+
+func TestContainsReservedKeywords_WithActualDenyList(t *testing.T) {
+	tests := []struct {
+		name     string
+		query    string
+		expected bool
+	}{
+		{
+			name:     "EXPLAIN statement should be detected",
+			query:    "EXPLAIN SELECT * FROM users",
+			expected: true,
+		},
+		{
+			name:     "SELECT with explain in string should not be detected",
+			query:    "SELECT 'explain this' FROM users",
+			expected: false,
+		},
+		{
+			name:     "PREPARE statement should be detected",
+			query:    "PREPARE stmt AS SELECT * FROM users WHERE id = $1",
+			expected: true,
+		},
+		{
+			name:     "SELECT with prepare in column name should not be detected",
+			query:    "SELECT prepare_date FROM users",
+			expected: false,
+		},
+		{
+			name:     "SET statement should be detected",
+			query:    "SET search_path TO public",
+			expected: true,
+		},
+		{
+			name:     "SELECT with set in string should not be detected",
+			query:    "SELECT 'set this value' FROM users",
+			expected: false,
+		},
+		{
+			name:     "SELECT with reserved word in quoted identifier",
+			query:    `SELECT "insert" FROM users`,
+			expected: false,
+		},
+		{
+			name:     "SELECT with reserved word in table alias",
+			query:    "SELECT * FROM users AS insert_table",
+			expected: false,
+		},
+		{
+			name:     "SELECT with legacy LOCK IN SHARE MODE",
+			query:    "SELECT name FROM users LOCK IN SHARE MODE",
+			expected: false,
+		},
+		{
+			name:     "SELECT with FOR UPDATE",
```

Comment:
My understanding is this will be dropped? I think it shouldn't, if that is possible.
We can always iterate of course.

Reply:
Yes, you're correct, this will currently be ignored. I'm looking into it now. I was under the impression from the tests that a compound command is properly identified, but that's not the case so.

For now, it may be best to continue to drop it, not because `FOR UPDATE` is a problem but because it contains `UPDATE` and we can not (easily) tell the difference.